appveyor.yml

```diff
@@ -0,0 +1,29 @@
+version: 1.0.{build}
+pull_requests:
+  do_not_increment_build_number: true
+branches:
+  except:
+  - coverity_scan
+  - gh_pages
+skip_tags: true
+image:
+  - Visual Studio 2017
+configuration:
+  - Debug
+  - Release
+platform:
+  - x64
+  - Win32
+init:
+  - cmd: |
+      cmake -version
+before_build:
+  - cmd: |
+      cd /d %APPVEYOR_BUILD_FOLDER%
+      if "%PLATFORM%"=="Win32" mkdir build & cd build & cmake -G"Visual Studio 15 2017" ..
+      if "%PLATFORM%"=="x64" mkdir build & cd build & cmake -G"Visual Studio 15 2017 Win64" ..

```

Comment:
Even if there is some conditional part on the platform part, there is a bit too much repetition here. As pointed out earlier.
```yaml
before_build:
  - cmd: |
      cd /d %APPVEYOR_BUILD_FOLDER%
      mkdir build
      cd build
      if "%PLATFORM%"=="Win32" cmake -G"Visual Studio 15 2017" ..
      if "%PLATFORM%"=="x64" cmake -G"Visual Studio 15 2017 Win64" ..
```
This does not repeat what does not need repeating.

BTW, why is the `cd` command necessary? Have you tried without it? Other cmake-based projects I've seen don't use this construct and work just fine.

Reply:
@akosthekiss  I am not sure you way it correct,  it's better to have appeyvor ci to verify it.

Reply:
using cmd is better if we want to add more functional in the future.